modules/core/shared/src/main/scala/lucuma/core/data/PerSite.scala

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2016-2025 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core.data
+
+import cats.Functor
+import cats.kernel.Eq
+import cats.kernel.Monoid
+import cats.syntax.monoid.*
+import io.circe.Decoder
+import io.circe.Encoder
+import io.circe.Json
+import io.circe.syntax.*
+import lucuma.core.enums.Site
+
+/** A total map from Site to A. */
+final case class PerSite[+A](gn: A, gs: A) extends (Site => A):
```

Comment:
Aren't `case class`es `final` by default in Scala 3?

Reply:
No, Martin has some cases where he likes to extend them.

```scala
scala> case class Foo()
// defined case class Foo
                                                                                
scala> class Bar extends Foo
// defined class Bar
```